npe2/manifest/sample_data.py

```diff
@@ -0,0 +1,49 @@
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from pydantic import BaseModel
+from pydantic.fields import Field
+
+from .._types import LayerData
+from .utils import Executable
+
+if TYPE_CHECKING:
+    from .._command_registry import CommandRegistry
+
+
+class _SampleDataContribution(BaseModel, ABC):
+    display_name: str
+    key: str  # python identifier
```

Comment:
maybe call this `python_name` instead?  Could also add regex validation similar to what was done for command.python_name.

Reply:
this one isn't actually a path to a python callable (like other `python_names` are) ... it's a key used to identify this sample programmatically ... for instance `viewer.open_sample('plugin_name', 'sample_key')` ... we don't want that to be the display name, since caps and spaces and parens and all that jazz are annoying

Reply:
added a description though to make that clearer